### ISSUE TITLE
Unexpected behavior while running into a non-terminating revert

### ISSUE THREAD

**Issue by hihiben** [trufflesuite/ganache-core#620]
Having an issue when performing a function call that resets the state unexpectedly while executing through ganache-cli.

* I asked about this on [gitter](https://gitter.im/ConsenSys/truffle): [x]
* Link to my question on gitter: https://gitter.im/ConsenSys/truffle?at=5f4784389566774dfe36c1e1

## Expected Behavior

Here's a simplified sample code that runs into the same problem.

```
pragma solidity ^0.5.0;

contract foo {
    function bar(bool flag) view public returns (bool) {
        require(flag, "error");
        return true;
    }
}

contract test {
    foo public f;
    bool public success;
    uint256 public count;

    constructor() public {
        f = new foo();
        count = 0;
    }

    function successNoRevert() public {
        count = 1;
        (success,) = address(f).staticcall(
            abi.encodeWithSignature(
                "bar(bool)", true
            )    
        );
        if (!success)
            count = count + 1;
        else
            count = count + 2;
    }

    function successRevert() public {
        count = 1;
        (success,) = address(f).staticcall(
            abi.encodeWithSignature(
                "bar(bool)", false
            )    
        );
        if (!success)
            count = count + 1;
        else
            count = count + 2;
    }

    function reset() public {
        count = 0;
    }
}
```

When calling `successRevert()` in `contract test`, `count` should be 2.

## Current Behavior

When calling `successRevert()` in `contract test`, `count` is 1 plus the current value.

## Possible Solution
Not a solution but might be the reason. 
`successRevert()` involves a reverting function call that does not terminate the entire transaction execution. However, it seems that the state was reverted to the initial state after executing.

## Steps to Reproduce (for bugs)
1. Deploy contract test
2. Call count() should return 0
3. Call successNoRevert()and call count() return 3
4. Call successRevert() and call count() return 4

## Context
We're testing by forking the mainnet and the results keep failing due to the behavior of 1inch contract.
Transactions like https://etherscan.io/tx/0x1d3403d4aa11e1edca3ee665c84dfdef19368eec3989d0d93f182e78d45b92a1 cannot behave correctly in this case.

## Your Environment
* Version used: v6.10.1
* Version of Truffle/Remix/Other tools used: v5.1.16
* NodeJS Version: [ ] 6.x, [ ] 7.x (unsupported), [ ] 8.x, [ ] 9.x, [x]10.x
* Operating System and version (include distro if Linux): Catalina 10.15.6
* I intend to submit a pull request to fix this issue: [ ]

**Comment by davidmurdoch** [comment]
Likely related to #611

**Comment by seesemichaelj** [comment]
Hey @hihiben, we released a fix to #611 that hopefully fixes your issue. Can you try that version and let us know if  it works or not?

ganache-cli: https://github.com/trufflesuite/ganache-cli/releases/tag/v6.12.0
ganache-core: https://github.com/trufflesuite/ganache-core/releases/tag/v2.13.0
